### PR TITLE
[Bogdan Pana] sequence_task_3

### PR DESCRIPTION
**Pull request by BogdanPana26**

### CODE REVIEW

File: sequence_task_3/student/include/impl.h [hunk 1]
```diff
@@ -8,45 +8,120 @@ class DynamicPlaylist
...
+    {
+        return m_tracklist.cend();
+    };
 
     DynamicPlaylist() = default;
 
     /** @todo Constructor from two iterators*/
-
+    template<class T>
+    DynamicPlaylist(T first, T second)
```
Comment (smay1613) [inline sequence_task_3/student/include/impl.h]: 1. Use initialization list
2. Use sequence constructor

File: sequence_task_3/student/include/impl.h [hunk 2]
```diff
@@ -8,45 +8,120 @@ class DynamicPlaylist
...
     /** @todo Set track to be played next from lvalue */
-    void playNext(const Song_t& song);
+    void playNext(const Song_t& song)
+    {
+        auto nextsong = begin();
+        if(hasTracks())
+        {
+            nextsong++;
+        }
+        m_tracklist.emplace(nextsong, song);
```
Comment (smay1613) [inline sequence_task_3/student/include/impl.h]: insert is more suitable here, as you insert lvalue

File: sequence_task_3/student/include/impl.h [hunk 3]
```diff
@@ -8,45 +8,120 @@ class DynamicPlaylist
...
+    void add(Args&&... songData)
+    {
+        m_tracklist.emplace_back(std::forward<Args>(songData)...);
+    }
 
     /** @todo Add track to the end of the queue from lvalue */
-    void add(const Song_t& song);
+    void add(const Song_t& song)
+    {
+        m_tracklist.emplace_back(song);
```
Comment (smay1613) [inline sequence_task_3/student/include/impl.h]: push_back
we're pushing already constructed value

File: sequence_task_3/student/include/impl.h (hunk 1 above)
Comment (BogdanPana26) [reply sequence_task_3/student/include/impl.h]: More details please. I don't understand what is wrong .

Comment (smay1613) [reply sequence_task_3/student/include/impl.h]: You may use the constructor directly and initialize container with two iterators (check "Containers common/Constructors" section of the training)
For that you may use member initializer list (https://www.learncpp.com/cpp-tutorial/constructor-member-initializer-lists/)

File: sequence_task_3/student/include/impl.h [hunk 4]
```diff
@@ -8,45 +8,119 @@ class DynamicPlaylist
...
+    const_iterator end() const
+    {
+        return m_tracklist.cend();
+    };
 
     DynamicPlaylist() = default;
 
     /** @todo Constructor from two iterators*/
+    template<class T>
+    DynamicPlaylist(T first, T second):m_tracklist{typename Container::template deque<Song_t>(first,second)}{}
```
Comment (smay1613) [inline sequence_task_3/student/include/impl.h]: why do you need that stuff? you can directly pass first and second to a tracklist member

Comment (BogdanPana26) [reply sequence_task_3/student/include/impl.h]: sorry I misunderstood what you meant first time  :))

Comment (smay1613) [reply sequence_task_3/student/include/impl.h]: just remove this:
typename Container::template deque<Song_t>()

**Pull request closed without merging**